### ISSUE TITLE
Execution failed for task ':wakelock:compileDebugKotlin'.

### ISSUE THREAD

**Issue by bhanuka96** [creativecreatorormaybenot/wakelock#52]
I got this error after upgrade to `^0.2.0+2`.

I found this solution to fix this,
  `wakelock: 0.2.0+1`

```
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (13, 50): 2 type arguments expected for class HashMap<K : Any!, V : Any!>
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (15, 33): Type mismatch: inferred type is String but Object was expected
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (16, 40): Unresolved reference: getClass
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (17, 33): Null can not be a value of a non-null type Object
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (32, 22): 2 type arguments expected for class HashMap<K : Any!, V : Any!>
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (34, 39): Type mismatch: inferred type is Boolean? but Object was expected
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (39, 30): 2 type arguments expected for class HashMap<K : Any!, V : Any!>
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (42, 40): Type mismatch: inferred type is Object but Boolean? was expected
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (59, 22): 2 type arguments expected for class HashMap<K : Any!, V : Any!>
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (61, 40): Type mismatch: inferred type is Boolean? but Object was expected
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (66, 30): 2 type arguments expected for class HashMap<K : Any!, V : Any!>
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (69, 41): Type mismatch: inferred type is Object but Boolean? was expected
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (84, 64): Type inference failed. Expected type mismatch: inferred type is BasicMessageChannel<Any!> but BasicMessageChannel<Object> was expected
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (84, 84): Type mismatch: inferred type is BinaryMessenger? but BinaryMessenger was expected
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (87, 58): 2 type arguments expected for class HashMap<K : Any!, V : Any!>
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (89, 118): 2 type arguments expected. Use 'HashMap<*, *>' if you don't want to pass type arguments
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (95, 41): Type mismatch: inferred type is HashMap<String, [ERROR : HashMap]> but Object? was expected
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (102, 64): Type inference failed. Expected type mismatch: inferred type is BasicMessageChannel<Any!> but BasicMessageChannel<Object> was expected
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (102, 84): Type mismatch: inferred type is BinaryMessenger? but BinaryMessenger was expected
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (105, 58): 2 type arguments expected for class HashMap<K : Any!, V : Any!>
e: /Users/userName/flutter/.pub-cache/hosted/pub.dartlang.org/wakelock-0.2.0+2/android/src/main/kotlin/creativemaybeno/wakelock/Messages.kt: (112, 41): Type mismatch: inferred type is HashMap<String, [ERROR : HashMap]> but Object? was expected

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':wakelock:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 18s
Exception: Gradle task assembleDevDebug failed with exit code 1


```

**Comment by creativecreatorormaybenot** [comment]
@bhanuka96 Can you try doing `flutter clean` and then getting the new version?

**Comment by bhanuka96** [comment]
@creativecreatorormaybenot I did 'flutter clean'. this is happening to my all projects.

```
Flutter 1.22.2 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 84f3d28555 (9 days ago) • 2020-10-15 16:26:19 -0700
Engine • revision b8752bbfff
Tools • Dart 2.10.2
```

**Comment by creativecreatorormaybenot** [comment]
@bhanuka96 I think that the issue is fixed 🎉 

Can you confirm this by testing `0.2.1`?
Thanks a lot for your help 🙂

**Comment by mehmetartun** [comment]
> @bhanuka96 I think that the issue is fixed 🎉
> 
> Can you confirm this by testing `0.2.1`?
> Thanks a lot for your help 🙂

Thank you I was having same issue and 0.2.1 fixed it!

**Comment by creativecreatorormaybenot** [comment]
@mehmetartun Thanks for the response!

I will close this issue then :)

**Comment by creativecreatorormaybenot** [comment]
To fix the `wakelock: 0.2.0+2` "Execution failed for task ':wakelock:compileDebugKotlin'." issue on Android, upgrade the wakelock plugin:

```yaml
wakelock: ^0.2.1
```

I will lock this issue - if you experience a similar or any other problem, please open a new issue 🙂
